backend/YouTubeV2.Application/Services/UserService.cs

```diff
@@ -15,12 +17,33 @@ public class UserService
         private readonly UserManager<User> _userManager;
         private readonly IBlobImageService _blobImageService;
         private readonly RegisterDtoValidator _registerDtoValidator;
+        private readonly LoginDtoValidator _loginDtoValidator;
+        private readonly JwtHandler _jwtHandler;
 
-        public UserService(UserManager<User> userManager, IBlobImageService blobImageService, RegisterDtoValidator registerDtoValidator)
+        public UserService(UserManager<User> userManager, IBlobImageService blobImageService,
+            RegisterDtoValidator registerDtoValidator, LoginDtoValidator loginDtoValidator, JwtHandler jwtHandler)
         {
             _userManager = userManager;
             _blobImageService = blobImageService;
             _registerDtoValidator = registerDtoValidator;
+            _loginDtoValidator = loginDtoValidator;
+            _jwtHandler = jwtHandler;
+        }
+
+        public async Task<LoginResponseDto> LoginAsync(LoginDto loginDto, CancellationToken cancellationToken)
+        {
+            await _loginDtoValidator.ValidateAndThrowAsync(loginDto, cancellationToken);
+
+            User user = await GetUserByEmailAsync(loginDto.email);
+
+            if (!await _userManager.IsEmailConfirmedAsync(user))
```

Comment:
we don't confirm emails. Its not in the specification

Reply:
deleted